File: src/main/java/com/example/lms/service/EmployeeService.java
package com.example.lms.service;

import com.example.lms.entity.Employee;

import java.util.List;

public interface EmployeeService {
    List<Employee> getAllEmployees();

    Employee getEmployeeById(int employeeId);

    Employee saveEmployee(Employee employee);

    Employee updateEmployee(int employeeId, Employee employee);

    void deleteEmployee(int employeeId);
}

File: leavemanagement1ui/src/services/EmployeeService.js
import axios from 'axios';

const EMPLOYEE_API_BASE_URL = 'http://localhost:8089/api/employees';


class EmployeeService {
  getAllEmployees() {
    return axios.get(EMPLOYEE_API_BASE_URL);
  }

  getEmployeeById(employeeId) {
    return axios.get(`${EMPLOYEE_API_BASE_URL}/${employeeId}`);
  }

  saveEmployee(employee) {
    return axios.post(EMPLOYEE_API_BASE_URL, employee);
  }

  updateEmployee(employeeId, employee) {
    return axios.put(`${EMPLOYEE_API_BASE_URL}/${employeeId}`, employee);
  }

  deleteEmployee(employeeId) {
    return axios.delete(`${EMPLOYEE_API_BASE_URL}/${employeeId}`);
  }
}

export default new EmployeeService();
